crates/commons/src/message.rs

```diff
@@ -43,10 +44,38 @@ pub enum Message {
     },
     TradeError {
         order_id: Uuid,
-        error: String,
+        error: TradingError,
     },
 }
 
+#[derive(Serialize, Deserialize, Clone, Error, Debug, PartialEq)]
+pub enum TradingError {
+    #[error("Invalid order: {0}")]
+    InvalidOrder(String),
+    #[error("{0}")]
+    NoMatchFound(String),
+    #[error("{0}")]
```

Comment:
:wrench: We can use `#[error(transparent)]` here.

Reply:
I don't think this is possible here, because the error must be serializable and `anyhow::Error` is not.

Reply:
Ah, sorry. It used to be `String` earlier in the diff.